common/models/User-Identity.js

```diff
@@ -145,27 +147,42 @@ export default function(UserIdent) {
         user.picture = picture;
         userChanged = true;
       }
-
       if (!githubRegex.test(provider) && profile) {
         user[provider] = getUsernameFromProvider(provider, profile);
         userChanged = true;
+        if (userChanged) {
+          return user.save(function(err) {
+            if (err) { return next(err); }
+            return next();
+          });
+        }
+        debug('exiting after user identity before save');
+        return next();
       }
 
       // if user signed in with github refresh their info
       if (githubRegex.test(provider) && profile && profile._json) {
-        debug("user isn't github cool or username from github is different");
-        setProfileFromGithub(user, profile, profile._json);
-        userChanged = true;
-      }
-
+        checkRepoAccess(userIdent.credentials.accessToken, function(bool) {
+          if (bool) {
+            debug('User isGithubRepoCool');
+            setProfileFromRepoAccess(user, profile, profile._json);
+            userChanged = true;
+          } else {
+            debug("user isn't github repo cool");
+            setProfileFromGithub(user, profile, profile._json);
+            userChanged = true;
+          }
 
-      if (userChanged) {
-        return user.save(function(err) {
-          if (err) { return next(err); }
+          if (userChanged) {
+            return user.save(function(err) {
+              if (err) { return next(err); }
+              return next();
```

Comment:
@BerkeleyTrue it returns next here, then if the user hasnt changed, returns next, and there is a final one at the end incase that async function fails